File: MetodosOrdenamiento/Radix/radix.py
import random

def get_digit(num, position):
    for _ in range(position):
        num //= 10
    return num % 10

def radix_sort(arr):
    if not arr:
        return arr
    
    max_num = max(arr)
    max_digits = 0

    while max_num > 0:
        max_digits += 1
        max_num //= 10

    for digit in range(max_digits):
        buckets = [[] for _ in range(10)]

        for num in arr:
            current_digit = get_digit(num, digit)
            buckets[current_digit].append(num)

        arr = [num for bucket in buckets for num in bucket]

    return arr

arr = [random.randint(1,100) for _ in range(10)]

print("Arreglo original:"," ".join(map(str, arr)))

sorted_arr = radix_sort(arr.copy())

print("Arreglo ordenado:"," ".join(map(str, sorted_arr)))

File: MetodosOrdenamiento/Radix/radix.js
function getDigit(num, position){
    for(let i=0; i<position;i++){
        num=Math.floor(num/10);
    }
    return num % 10;
}

function radixSort(arr){
    if(arr.length === 0)return arr;

    let maxNum = Math.max(...arr);
    let maxDigits = 0;

    while (maxNum > 0){
        maxDigits++;
        maxNum = Math.floor(maxNum / 10);
    }
    for(let digit = 0; digit < maxDigits; digit++){
        let buckets = Array.from({length: 10}, () => []);

        for(let num of arr){
            let currentDigit = getDigit(num, digit);
            buckets[currentDigit].push(num);
        }
        arr = [].concat(...buckets);
    }
    return arr;
}
let arr = Array.from({length:10}, () => Math.floor(Math.random()  * 101));

console.log("Arreglo original: ", arr.join(" "));

let sortedArr = radixSort([...arr]);

console.log("Arreglo ordenado:", sortedArr.join(" "));